src/tabset/tabset.ts

```diff
@@ -19,27 +21,43 @@ export class NgbTabContent {
   constructor(public templateRef: TemplateRef<any>) {}
 }
 
+/**
+ * A directive representing an individual tab.
+ */
 @Directive({selector: 'ngb-tab'})
 export class NgbTab {
+  /**
+   * Unique tab identifier. Must be unique for the entire document for proper accessibility support.
+   */
+  @Input() id: string = `ngb-tab-${nextId++}`;
+  /**
+   * Simple (string only) title. Use the "NgbTabTitle" directive for more complex use-cases.
+   */
   @Input() title: string;
+  /**
+   * Allows toggling disabled state of a given state. Disabled tabs can't be selected.
+   */
   @Input() disabled: boolean = false;
 
   @ContentChild(NgbTabContent) contentTpl: NgbTabContent;
   @ContentChild(NgbTabTitle) titleTpl: NgbTabTitle;
 }
 
+/**
+ * A component that makes it easy to create tabbed interface.
+ */
 @Component({
   selector: 'ngb-tabset',
   template: `
     <ul class="nav nav-tabs" role="tablist">
-      <li class="nav-item" *ngFor="let tab of tabs; let i=index">
-        <a class="nav-link" [class.active]="i === activeIdx" [class.disabled]="tab.disabled" (click)="select(i)">
+      <li class="nav-item" *ngFor="let tab of tabs">
+        <a [id]="tab.id" class="nav-link" [class.active]="tab.id === activeId" [class.disabled]="tab.disabled" (click)="select(tab.id)">
```

Comment:
in the UIBS version we pass the event object to the click handler so that the user can cancel.  i don't think we need it for the initial release, but know that someone will ask for it as some point.